splice_machine/src/main/java/com/splicemachine/derby/stream/function/QuoteTrackingTokenizer.java

```diff
@@ -30,18 +33,86 @@
  * versus columns with values {@code "Foo"}.
  *
  * @author Scott Fines
- *         Date: 9/29/16
+ * Date: 9/29/16
  */
-public class QuoteTrackingTokenizer extends AbstractTokenizer{
+public class QuoteTrackingTokenizer extends AbstractTokenizer {
+
+    private static final Logger LOG=Logger.getLogger(QuoteTrackingTokenizer.class);
+
+    private final List<Integer> valueSizeHints;
+    // the line number where a potential multi-line cell starts
+    private int potentialSpaces;
+    private int charIndex;
+    private boolean wasQuoted;
+    // the line number where a potential multi-line cell starts
+    private int quoteScopeStartingLine;
+    private TokenizerState state;
+    private int colIdx;
+    private boolean firstRun;
+    private boolean checkExactSizeFirst = false;
+    private int rowIdx = 0;
+    private int[] exactColumnSizes = null;
+    final static int SCAN_THRESHOLD = 10000;
```

Comment:
If you want to add a configuration option:

1. You'd get the config option with SIDriver.driver().getConfiguration().getCsvScanThreshold() (or whatever name you come up with)
2. You need to add that method to the interface, implementation and ConfigurationBuilder
3. Add the configuration to a *Configuration class (possibly SQLConfiguration, that's where `importMaxQuotedColumnLines` is defined)

Reply:
done. Thanks for the explanation.